### DISCUSSION TITLE
What am I working on...

### DISCUSSION THREAD

**Discussion by palexdev** [palexdev/MaterialFX#349] [Show and tell]
Hello everyone, I know it's only Thursday, but I felt like giving you a quick recap of what I am working on this week.
- I made some improvements to the Theming API on the `main` branch. I'm now waiting for @infinite-dev22 to make the requested changes to #325. If too much time passes, I'll do it personally and release a new version, so don't worry, improvements and bug fixes will come soon. (P.S: for those who want to try it in preview, there's a jar here #345)
- I just released a small but important update on the `rewrite` branch, which improves the `MomentumTransition`'s behavior
- I've been working on this for several weeks now, it's not MaterialFX, but it's strictly related. I'm talking about [VirtualizedFX](https://github.com/palexdev/VirtualizedFX). Just as I'm doing for MaterialFX here, I'm rewriting it, in fact there is now a `rewrite` branch there too (currently empty). @stefanofornari remember when you told me I should have considered TDD (here [#4](https://github.com/palexdev/VirtualizedFX/issues/4)), you were damn right. <details><summary>Spoiler</summary><image src="https://github.com/palexdev/MaterialFX/assets/16880178/afe1c1bb-a05a-49a5-8cf0-0228ab88ac18"></image></details> This rewrite also aims to simplify things, while still trying to be as performant as possible. For now, the only container I've almost finished is the `VirtualizedList` (previously VirtualFlow), if all goes well, it should be available on GitHub by the end of the week.

I know, it may not seem like a lot, but trust me, working on VirtualizedFX is **hard** and takes a lot of time.
As always, stay tuned, have a nice day, and good luck with your projects 😉

**Comment by stefanofornari** [comment]
Thanks for your restless effort! I look forward to the outcome, have a great weekend!

**Comment by palexdev** [comment]
First version of [VirtualizedFX](https://github.com/palexdev/VirtualizedFX/tree/rewrite) from the rewrite branch is now out 💪
